modules/web/src/de/diedavids/cuba/wizard/web/gui/components/WebWizardStep.java

```diff
@@ -1,27 +1,56 @@
 package de.diedavids.cuba.wizard.web.gui.components;
 
+import com.haulmont.cuba.gui.components.Component;
+import com.haulmont.cuba.gui.components.TabSheet;
 import com.haulmont.cuba.web.gui.components.WebVBoxLayout;
 import de.diedavids.cuba.wizard.gui.components.WizardStep;
 import de.diedavids.cuba.wizard.gui.components.WizardStepAware;
 
 public class WebWizardStep extends WebVBoxLayout implements WizardStep {
+    private String name;
+    private WizardStepAware stepComponent;
+    private TabSheet.Tab tabComponent;
+    private String icon;
+
     public WebWizardStep() {
+
     }
 
+    public WebWizardStep(String name, WizardStepAware stepComponent) {
+        this.name = name;
+        this.stepComponent = stepComponent;
+        this.add((Component) stepComponent);
+    }
+
+
     @Override
     public void onActivate() {
-        getWizardStepAware().onActivate();
+        WizardStepAware wizardStepAware = getWizardStepAware();
+        if (wizardStepAware != null) {
+            wizardStepAware.onActivate();
+        }
     }
 
 
     @Override
     public boolean preClose() {
-        return getWizardStepAware().preClose();
+        WizardStepAware wizardStepAware = getWizardStepAware();
+        if (wizardStepAware != null) {
+            return wizardStepAware.preClose();
+        }
+        else {
+            return false;
+        }
     }
 
 
 
     private WizardStepAware getWizardStepAware() {
-        return (WizardStepAware) ownComponents.get(0);
+        if (ownComponents.size() > 0) {
```

Comment:
i'm not really sure why this is necessary anymore. I think it has to do with how it is loaded here: https://github.com/mariodavid/cuba-component-wizard/blob/master/modules/gui/src/de/diedavids/cuba/wizard/gui/xml/layout/loaders/WizardStepLoader.java#L7